src/Gui/DlgPreferencesImp.cpp

```diff
@@ -67,71 +69,103 @@ DlgPreferencesImp::DlgPreferencesImp(QWidget* parent, Qt::WindowFlags fl)
+    ui->tabWidgetStack->addWidget(tabWidget);
+    tabWidget->setProperty("GroupName", QVariant(groupNameQString));
+
+    QListWidgetItem* item = new QListWidgetItem(ui->listBox);
+    item->setData(GroupNameRole, QVariant(groupNameQString));
+    item->setText(QObject::tr(groupNameQString.toLatin1()));
+    item->setToolTip(QObject::tr(groupNameQString.toLatin1()));
+    std::string fileName = groupName;
+    for (auto & ch : fileName) {
+        if (ch == ' ') ch = '_';
+        else ch = tolower(ch);
+    }
+    fileName = std::string("preferences-") + fileName;
+    QPixmap icon = Gui::BitmapFactory().pixmapFromSvg(fileName.c_str(), QSize(48, 48));
+    if (icon.isNull()) {
+        icon = Gui::BitmapFactory().pixmap(fileName.c_str());
         if (icon.isNull()) {
-            icon = Gui::BitmapFactory().pixmap(fileName.c_str());
-            if (icon.isNull()) {
-                qWarning() << "No group icon found for " << fileName.c_str();
-            }
-            else if (icon.size() != QSize(96,96)) {
-                qWarning() << "Group icon for " << fileName.c_str() << " is not of size 96x96";
-            }
+            qWarning() << "No group icon found for " << fileName.c_str();
         }
-        item->setIcon(icon);
-        item->setTextAlignment(Qt::AlignHCenter);
-        item->setFlags(Qt::ItemIsSelectable | Qt::ItemIsEnabled);
-        for (std::list<std::string>::iterator jt = it->second.begin(); jt != it->second.end(); ++jt) {
-            PreferencePage* page = WidgetFactory().createPreferencePage(jt->c_str());
-            if (page) {
-                tabWidget->addTab(page, page->windowTitle());
-                page->loadSettings();
-            }
-            else {
-                Base::Console().Warning("%s is not a preference page\n", jt->c_str());
-            }
+        else if (icon.size() != QSize(48, 48)) {
+            icon = Gui::BitmapFactory().resize(48, 48, icon, Qt::TransparentMode);
+            qWarning() << "Group icon for " << fileName.c_str() << " is not of size 48x48, so it was scaled";
         }
     }
+    item->setIcon(icon);
+    item->setTextAlignment(Qt::AlignHCenter);
+    item->setFlags(Qt::ItemIsSelectable | Qt::ItemIsEnabled);
 
-    // show the first group
-    ui->listBox->setCurrentRow(0);
+    return tabWidget;
+}
+
+/**
+ * Create a new preference page called \a pageName on the group tab \a tabWidget.
+ */
+void DlgPreferencesImp::createPageInGroup(QTabWidget *tabWidget, const std::string &pageName)
+{
+    PreferencePage* page = WidgetFactory().createPreferencePage(pageName.c_str());
+    if (page) {
+        tabWidget->addTab(page, page->windowTitle());
+        page->loadSettings();
+        page->setProperty("GroupName", tabWidget->property("GroupName"));
+        page->setProperty("PageName", QVariant(QString::fromStdString(pageName)));
+    }
+    else {
+        Base::Console().Warning("%s is not a preference page\n", pageName);

```

Comment:
Strange that the gcc doesn't spot this bug because it used to do so. I found it with clang the printed this error:
`cannot pass object of non-trivial type 'const std::string' (aka 'const basic_string<char>') through variadic method; call will abort at runtime [-Wnon-pod-varargs]`